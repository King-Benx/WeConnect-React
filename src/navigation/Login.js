import React from 'react';
import { Link, BrowserRouter } from 'react-router-dom';

const Login = ()=>(
    <li>
        <BrowserRouter >
        <Link to='/login'>
            <b><i className="glyphicon glyphicon-log-in"></i> Login</b>
        </Link>
        </BrowserRouter>  
    </li>
)
export default Login;
