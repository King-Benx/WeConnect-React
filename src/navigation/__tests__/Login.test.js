import React from 'react';
import { shallow } from 'enzyme';
import Login from "../Login";
import renderer from 'react-test-renderer';

describe('These are tests the login link', () => {
    const event={
        target:{
            value:{}
        },
        preventDefault: () => {

        }
    }
    let mountedLoginLink;

    beforeEach(()=>{
        mountedLoginLink= shallow(<Login />);
    });

    it('renders without crashing', () => {
        mountedLoginLink =  shallow(<Login />)
        const tree = renderer.create(<Login />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('has links', () =>{
        const links = mountedLoginLink.find('Link')
        expect(links.length).toBe(1)
    });

    it('has a list', () =>{
        const lists = mountedLoginLink.find('li')
        expect(lists.length).toBe(1)
    });

});