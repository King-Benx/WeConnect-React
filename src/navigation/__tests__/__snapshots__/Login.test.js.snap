// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`These are tests the login link renders without crashing 1`] = `
<li>
  <a
    href="/login"
    onClick={[Function]}
  >
    <b>
      <i
        className="glyphicon glyphicon-log-in"
      />
       Login
    </b>
  </a>
</li>
`;
